tools/make.py

```diff
@@ -173,6 +186,11 @@
                       default=False,
                       help="Link with mbed test library")
 
+    parser.add_argument("--metadata",
```

Comment:
I realize this is bikeshedding,  so feel free to shut this down immediately. Is `metadata` really a descriptive enough name? Perhaps this could be `--build-data` or `--json-report`? `--metadata` almost sounds like it could affect the build, which may be confusing with `--app-config`.